src/commands/games/leagueoflegends/rotation.js

```diff
@@ -1,34 +1,36 @@
-const { Command, SwitchbladeEmbed, CommandError } = require('../../../')
-
-module.exports = class LeagueOfLegendsRotation extends Command {
-  constructor (client) {
-    super({
-      name: 'rotation',
-      aliases: ['r'],
-      parent: 'leagueoflegends',
-      parameters: [[{
-        type: 'booleanFlag', name: 'newplayers', aliases: ['np']
-      }]]
-    }, client)
-  }
-
-  async run ({ t, author, channel, language, flags }) {
-    channel.startTyping()
-    const embed = new SwitchbladeEmbed(author)
-    try {
-      const rotation = await this.client.apis.lol.fetchChampionRotation()
-      const champions = flags['newplayers'] ? rotation.freeChampionIdsForNewPlayers : rotation.freeChampionIds
-      const championPayload = await await Promise.all(champions.map(async c => {
-        var payload = await this.client.apis.lol.fetchChampion(c, language, true)
-        return `**${payload.name}**, ${payload.title}`
-      }))
-      embed.setColor(this.parentCommand.embedColor)
-        .setAuthor(t(this.parentCommand.authorString), this.parentCommand.authorImage, this.parentCommand.authorURL)
-        .setTitle(t('commands:leagueoflegends.subcommands.rotation.weeklyChampRotation'))
-        .setDescription(championPayload.join('\n'))
-      channel.send(embed).then(() => channel.stopTyping())
-    } catch (e) {
-      throw new CommandError(t('errors:generic'))
-    }
-  }
-}
+const { Command, SwitchbladeEmbed, CommandError } = require('../../../')
+
+module.exports = class LeagueOfLegendsRotation extends Command {
+  constructor (client) {
+    super({
+      name: 'rotation',
+      aliases: ['r'],
+      parent: 'leagueoflegends',
+      parameters: [[{
+        type: 'booleanFlag', name: 'newplayers', aliases: ['np']
+      }]]
+    }, client)
+  }
+
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
+    const { fetchChampionRotation, fetchChampion } = this.client.apis.lol
+    try {
+      const {freeChampionIdsForNewPlayers, freeChampionIds} = await fetchChampionRotation()
+      const champions = flags['newplayers'] ? freeChampionIdsForNewPlayers : freeChampionIds
+      const championPayload = await Promise.all(champions.map(async c => {
+        var payload = await fetchChampion(c, language, true)

```

Comment:
use const and destruct!

```suggestion
        const { name, title } = await fetchChampion(c, language, true)
```